src/Link.tsx

```diff
@@ -55,24 +76,35 @@ class Link extends React.Component<LinkProps, undefined> {
         pushUrl(to);
       }
     }
-  }
+  };
 
   public render(): JSX.Element {
-    return (
-      <a
-        className={this.props.className}
-        href={this.props.to}
-        onClick={this.handleClick}
-      >
-        {this.props.children}
-      </a>
+    const {
+      innerRef,
+      pushUrl,
+      replace,
+      replaceUrl,
+      to,
+      ...props
+    } = this.props;
+
+    invariant(
+      this.context.router,
+      'You should not use <Link> outside a <Router>'
+    );
+
+    const href = this.context.router.history.createHref(
+      typeof to === 'string' ? { pathname: to } : to
     );
+
+    return <a {...props} onClick={this.handleClick} href={href} ref={innerRef}/>
```

Comment:
I think this would probably make a API docs update necessary.

Reply:
Good point, I'll update docs. I'm not the native speaker so feel free to fix anything in them that looks not correct or not clear to you.